mbed_tools_lib/python_helpers.py

```diff
@@ -0,0 +1,23 @@
+"""Helpers for python language related functions."""
+import sys
+from collections import namedtuple
+from typing import NamedTuple, Iterable, Any, cast
+
+
+def is_python_more_recent(major: int, minor: int) -> bool:
+    """States whether the python in use is more recent than set major,minor versions."""
+    return sys.version_info.major >= major and sys.version_info.minor >= minor
+
+
+def named_tuple_with_defaults(typename: str, field_names: Iterable[str], defaults: Iterable[Any]) -> NamedTuple:
```

Comment:
I'm not sure why we would need this. Using `dataclasses` as we have elsewhere solves the defaults problem. I'm not sure this makes sense as a library function.

Reply:
As we decided, since namedtuples are used in devices (Windows adapter), the utility and tests will be moved to this lib. However, this should be removed when no longer used elsewhere

Reply:
Why is something used only in one place extracted to the common lib? Since the introduction of `dataclasses` I can't see us needing this in the future - am I missing some crucial information? 
I'm slightly worried there's now going to be a _hard_ to track dependency between `mbed-devices` and `lib`, which we might've not needed in the first place. 🤔

Reply:
FYI, we wanted to avoid using namedtuples in the future, as they implement protocols that are not suitable for our usage of _value objects_.